backend/utils/hf_progress.py

```
@@ -12,28 +12,53 @@
+    *correct* tracker instance happens via a thread-local — each download
+    runs its blocking HF calls on its own thread, so the tqdm bars it creates
+    always resolve back to that thread's tracker, never another thread's.
+    """
+
+    _install_lock = threading.Lock()
+    _refcount = 0
+    _local = threading.local()
+
+    _original_tqdm_class = None
+    _original_tqdm_auto = None
+    _patched_modules: dict = {}
```

Ruff (0.15.20)

[warning] Mutable default value for class attribute

(RUF012)